genet/schedule_elements.py

```diff
@@ -3587,29 +3595,74 @@ def write_to_json(self, output_dir):
             json.dump(self.to_json(), outfile)
         self.write_extras(output_dir)
 
-    def write_to_geojson(self, output_dir, epsg):
+    def write_spatial(
+        self, output_dir, epsg: str = None, to_parquet=False, to_geojson=False, to_shp=False
+    ):
         """
-        Writes Schedule graph to nodes and edges geojson files.
+        Transforms Schedule to geopandas.GeoDataFrame of nodes and links and saves to
+        the given format to the requested file format(s)
         :param output_dir: output directory
-        :param epsg: projection if the geometry is to be reprojected, defaults to own projection
-        :return:
-        """
+        :param epsg: projection, if the geometry is to be reprojected, defaults to own projection
+        :param to_parquet: whether to save to parquet file format, defaults to no
+        :param to_geojson: whether to save to geojson file format, defaults to no
+        :param to_shp: whether to save to shape file format, defaults to no
+        :return: None, files are saved to disk
+        """
+        if not any((to_parquet, to_geojson, to_shp)):
```

Comment:
Again, I'd defer this to `save_geodataframe` so that you dont' have the repetition.